atom/browser/api/lib/web-contents.js

```diff
@@ -106,16 +112,37 @@ let wrapWebContents = function(webContents) {
     };
   }
 
+  for (let method of asyncWebFrameMethods) {
+    webContents[method] = function() {
+      let args = Array.prototype.slice.call(arguments);
+      this.send('ELECTRON_INTERNAL_RENDERER_ASYNC_WEB_FRAME_METHOD', method, args);
+    };
+  }
+
   // Make sure webContents.executeJavaScript would run the code only when the
   // webContents has been loaded.
   const executeJavaScript = webContents.executeJavaScript;
-  webContents.executeJavaScript = function(code, hasUserGesture) {
+  webContents.executeJavaScript = function(code, hasUserGesture, callback) {
+    let requestId = getNextId();
+    if (typeof hasUserGesture === "function") {
+      callback = hasUserGesture;
+      hasUserGesture = false;
+    }
+    if (callback !== null)
```

Comment:
I think we should check `undefined`? Probably needs to add a test case for `webContents.executeJavaScript`.